### PR TITLE
Chat & Channels : front

### PR DESCRIPTION
**Pull request by Ptitjen**
Conversation and Channel messages display
Added links to user's profile from avatars in Channel
Added GetProfile query to start profile page

### CODE REVIEW

File: apps/client/src/sidebar/pages/channel.tsx [hunk 1]
```diff
@@ -1,35 +1,156 @@
...
+    id: number;
+    name: string;
+    avatar: string;
+  }[];
+}) => {
+  const navigate = useNavigate();
+
+  return (
+    <div className="ml-24 flex h-6 w-full items-end justify-start">
+      <div className="mb-px mr-1 flex w-full justify-end text-end text-xs text-slate-300 ">
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Check the classes, it probably shouldn't be a flex container

File: apps/client/src/sidebar/pages/channel.tsx [hunk 2]
```diff
@@ -1,35 +1,156 @@
...
+    content: string;
+    sentAt: number;
+  };
+}) => {
+  const navigate = useNavigate();
   const getDate = (time: number): Date => {
     return new Date(time);
   };
+  return (
+    <div className="">
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Extra div?

File: apps/client/src/sidebar/pages/channel.tsx [hunk 3]
```diff
@@ -1,35 +1,156 @@
...
+      <div className="mt-6 flex justify-center text-center text-xs text-slate-300">
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Probably not a flex container

File: apps/client/src/sidebar/pages/channel.tsx [hunk 4]
```diff
@@ -1,35 +1,156 @@
...
+      <div className="flex w-full flex-row">
+        <div className="flex w-9 shrink-0 justify-center">
+          <div className="flex justify-center self-end">
+            <img
+              className="flex h-6 w-6 basis-1 self-center rounded-full transition-all hover:h-7 hover:w-7"
+              src={message.author.avatar}
+              onClick={() => navigate(`/profile/${message.author.id}`)}
+            />
+          </div>
+        </div>
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Clean the classes and remove useless elements

File: apps/client/src/sidebar/pages/channel.tsx [hunk 5]
```diff
@@ -1,35 +1,156 @@
...
+      </div>
+      <div className="flex flex-row ">
+        {
+          <ReadBy
+            users={message.readBy.map((Users) => {
+              return Users.user;
+            })}
+          />
+        }
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Extra '{}' ??

File: apps/client/src/sidebar/pages/channel.tsx [hunk 6]
```diff
@@ -1,35 +1,156 @@
...
-        <div>Owner: {data?.channel.owner.name}</div>
+  if (isLoading) {
+    return <div>Loading ...</div>;
+  }
+  if (isFetching) {
+    return <div>Fetching</div>;
+  }
+  if (error) {
+    return <div>Error</div>;
+  } else {
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx:129]: We should probably have real loading and error page/components at some time

File: apps/client/src/sidebar/pages/channel.tsx [hunk 7]
```diff
@@ -1,35 +1,156 @@
...
+            </div>
+          </div>
+        );
+      })}
+    </div>
+  );
+};
+
+const ChannelMessage = ({
+  message,
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Remember to use object destructuring

File: apps/client/src/sidebar/pages/channel.tsx [hunk 8]
```diff
@@ -1,35 +1,156 @@
...
+  } else {
+    return (
+      <div className="flex flex-col">
+        <div className="mt-4 flex w-full flex-col items-center justify-center border-2 border-black p-2 text-center text-sm">
+          <div>Channel: {data?.channel.name}</div>
+          <div>Owner: {data?.channel.owner.name}</div>
+        </div>
+        <div className="flex h-full w-full flex-col pr-2 pl-px">
+          {data?.channel.messages?.map((m, index) => (
+            <ChannelMessage key={index} message={m} />
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/channel.tsx]: Remember to use object destructuring

File: apps/client/src/sidebar/pages/chat.tsx [hunk 9]
```diff
@@ -1,37 +1,117 @@
...
+    };
+  };
+  userId: number;
+}) => {
+  const getDate = (time: number): Date => {
+    return new Date(time);
+  };
+  return (
+    <li className=" mx-2 mb-5 flex flex-col ">
+      <div className="mb-2 flex justify-center text-center text-xs text-slate-300">
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/chat.tsx]: Probably not a flex container

File: apps/client/src/sidebar/pages/chat.tsx [hunk 10]
```diff
@@ -1,37 +1,117 @@
...
+          message.author.id === userId ? "justify-start" : "justify-end"
+        } flex flex-row`}
+      >
+        <img
+          className={`${
+            message.author.id === userId ? "order-first" : "order-last"
+          } m-1 mb-px h-6 w-6 basis-1 self-end rounded-full`}
+          src={message.author.avatar}
+        />
+        <div>
```
Comment (SebastienWae) [inline apps/client/src/sidebar/pages/chat.tsx:39]: Extra div?

**Review by SebastienWae — CHANGES_REQUESTED** [review]

File: apps/client/src/sidebar/pages/channel.tsx (hunk 6 above)
Comment (Ptitjen) [reply apps/client/src/sidebar/pages/channel.tsx:129]: Ok, I'm going to do this next

**Pull request merged**